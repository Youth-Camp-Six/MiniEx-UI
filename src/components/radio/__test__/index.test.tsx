/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { RadioGroup } from '..';

const radioDataList = [
  { label: 'Unknown', value: '0' },
  { label: 'man', value: '1' },
  { label: 'woman', value: '2' },
];

describe('test RadioGroup comonent', () => {
  it('radio snapshot test', () => {
    const wrapper = render(<RadioGroup value={1} options={radioDataList}></RadioGroup>);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  it('renders radio options and updates selected value', () => {
    const onChange = jest.fn();
    const wrapper = render(
      <RadioGroup value={1} options={radioDataList} onChange={onChange}></RadioGroup>
    );

    const option0 = wrapper.getByTestId('radio-0');
    const option1 = wrapper.getByTestId('radio-1');
    const option2 = wrapper.getByTestId('radio-2');

    expect(option0).not.toHaveClass('mi-radio-checked');
    expect(option1).toHaveClass('mi-radio-checked');
    expect(option2).not.toHaveClass('mi-radio-checked');
    expect(onChange).not.toHaveBeenCalled();

    option0.click();
    expect(option0).toHaveClass('mi-radio-checked');
    expect(option1).not.toHaveClass('mi-radio-checked');
    expect(option2).not.toHaveClass('mi-radio-checked');
    expect(onChange).toHaveBeenCalledWith('0');
  });
});
