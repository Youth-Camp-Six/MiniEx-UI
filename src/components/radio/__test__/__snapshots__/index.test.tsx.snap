// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test RadioGroup comonent radio snapshot test 1`] = `
<DocumentFragment>
  <div
    class="mi-flex mi-radio lew-radio-group-x lew-radio-group-medium mi-flex-x mi-flex-x-start mi-flex-y-center"
    style="gap: 15px;"
  >
    <label
      class="mi-radio 
    
    mi-radio-medium
    
    "
      data-testid="radio-0"
    >
      <div
        class="icon-radio-box"
      >
        <div
          class="icon-radio"
        />
      </div>
      <input
        style="display: none;"
        type="radio"
        value=""
      />
      <span
        class="mi-radio-label"
      >
         Unknown
      </span>
    </label>
    <label
      class="mi-radio 
    mi-radio-checked
    mi-radio-medium
    
    "
      data-testid="radio-1"
    >
      <div
        class="icon-radio-box"
      >
        <div
          class="icon-radio"
        />
      </div>
      <input
        checked=""
        style="display: none;"
        type="radio"
        value=""
      />
      <span
        class="mi-radio-label"
      >
         man
      </span>
    </label>
    <label
      class="mi-radio 
    
    mi-radio-medium
    
    "
      data-testid="radio-2"
    >
      <div
        class="icon-radio-box"
      >
        <div
          class="icon-radio"
        />
      </div>
      <input
        style="display: none;"
        type="radio"
        value=""
      />
      <span
        class="mi-radio-label"
      >
         woman
      </span>
    </label>
  </div>
</DocumentFragment>
`;
